flex-checks/src/test/java/org/sonar/flex/checks/FileHeaderCheckTest.java

```diff
@@ -120,11 +73,17 @@ private FileHeaderCheck createCheck(String format, boolean isRegularExpression)
   }
 
   private void assertHasIssue(FileHeaderCheck check, File file) {
-    CheckMessagesVerifier.verify(checkMessages(file, check)).next().atLine(null);
+    List<Issue> issues = issues(check, file);
+    assertThat(issues).hasSize(1);
+    assertThat(issues.get(0).line()).isNull();
   }
 
   private void assertNoIssue(FileHeaderCheck check, File file) {
-    CheckMessagesVerifier.verify(checkMessages(file, check)).noMore();
+    assertThat(issues(check, file)).isEmpty();
+  }
+
+  private List<Issue> issues(FileHeaderCheck check, File file) {
+    return check.scanFileForIssues(FlexVerifier.createContext(file));
```

Comment:
As far as I can tell, the `FlexVerifier.createContext(...)` method is only public because of this call. I'm really not sure that this is required, nor wanted. It also make things a bit less obvious to me on _how_ we should test rules.

All these tests are only validating that there will be, or not, an issue  **on the file file itself**, rather than on specific lines. Method `assertHasIssue(...)` is checking for a single issue with `null` line. 

Don't you think it would be better to add a method in `FlexVerifier` called `verifyIssueOnFile(File file, FlexCheck check, String expectedIssueMessageOnFile)` instead? 
`assertNoIssue()` should also be equivalent to calling `FlexVerifier.verifyNoIssue(...)`.

Reply:
I agree. Done in 8a01811.